src/Common/Core/Impl/Services/ServiceManager.cs

```diff
@@ -28,22 +28,33 @@ public class ServiceManager : IServiceManager {
         public virtual IServiceManager AddService<T>(T service, Type type = null) where T : class {
             _disposeToken.ThrowIfDisposed();
 
-            type = type ?? typeof(T);
+            type = type ?? service.GetType();
             Check.ArgumentNull(nameof(service), service);
-            Check.InvalidOperation(() => _s.TryAdd(type, service), $"Service of type {type} already exists");
+           _s.TryAdd(type, service); // Ignore if service added second time
```

Comment:
May be worth adding a check. Same instance is OK, new instance is probably a bug.